dino_catalog/dinoviewer.rb

```diff
@@ -0,0 +1,90 @@
+def view_home
+  puts @messages if @messages
+  @messages = ""
+  puts "Hello there Dinosaur Hunter! What would you like to do today??"
+  puts "--- Type 'view' to see your Dinosaurs."
+  puts "--- Type 'add' to add a Dinosaur."
+  puts "--- Type 'search' to pull up Dinosaur Search Functions."
+  puts "--- Type 'save' to save your Dinosaurs."
+  puts "--- Type 'exit' to exit the program."
+  input = gets.downcase.chomp
+  if input == 'save' || input == 'add' || input == 'search' || input == 'view'
+    controller_home(input)
+  elsif input == 'exit'
+    puts "Catch Ya Later, Dinosaur Hunter!!"
+    exit
+  else
+    puts "Sorry! That's not a valid option!"
+    view_home
+  end
+end
+
+def view_add_dino
+  puts @messages if @messages
+  @messages = ""
+  dino = {}
+  puts "What is the name of your Dinosaur"
+  dino[:name] = gets.chomp
+  puts "What prehistoric era did your Dinosaur live?"
+  dino[:period] = gets.chomp
+  puts "Upon what continent(s) did your Dinosaur live?"
+  dino[:continent] = gets.chomp
+  puts "What kind of diet did your Dinosaur have?"
+  dino[:diet] = gets.chomp
+  puts "How heavy was this Dinosaur (in lbs)?"
+  dino[:weight_in_lbs] = gets.chomp
+  puts "Was your Dinosaur a (Biped) or a (Quadruped)?"
+  dino[:walking] = gets.chomp.capitalize!
+  puts "Write up a little Description if you'd like..."
+  dino[:description] = gets.chomp
+  puts "Here's your Dinosaur:"
+  dino.each { |k, v| puts "#{k}: #{v}" }
+  puts "Would you like to store it in your DinoDex? (Y/n)"
+  response = gets.downcase.chomp
+  controller_add_dino(dino, response)
```

Comment:
The view is invoking controller logic, and the controller is invoking view logic. Circular references are usually overly complex and confusing.